core/transactions-factories/token_management_transactions_factory.md

```diff
@@ -97,90 +97,164 @@ class TokenManagementTransactionsFactory:
-        hash: str;
+        hash: string;
         attributes: bytes;
-        uris: List[str];
+        uris: List[string];
     }): Transaction;
 
     create_transaction_for_pausing({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
     }): Transaction;
 
     create_transaction_for_unpausing({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
     }): Transaction;
 
     create_transaction_for_freezing({
         sender: IAddress;
         user: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
     }): Transaction;
 
     create_transaction_for_unfreezing({
         sender: IAddress;
         user: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
     }): Transaction;
 
     create_transaction_for_wiping({
         sender: IAddress;
         user: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
     }): Transaction;
 
     create_transaction_for_local_minting({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
         supplyToMint: int;
     }): Transaction;
 
     create_transaction_for_local_burning({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
         supplyToBurn: int;
     }): Transaction;
 
     create_transaction_for_updating_attributes({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
         tokenNonce: int;
         attributes: bytes;
     }): Transaction;
 
     create_transaction_for_adding_quantity({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
         tokenNonce: int;
         quantityToAdd: int;
     }): Transaction;
 
     create_transaction_for_burning_quantity({
         sender: IAddress;
-        tokenIdentifier: str;
+        tokenIdentifier: string;
         tokenNonce: int;
         quantityToBurn: int;
     }): Transaction;
 
-    ...
+    // receiver is the same as the sender
+    create_transaction_for_modifying_royalties({
+        sender: IAddress;
+        tokenIdentifier: string;
+        tokenNonce: int;
+        royalties: int;
+    }): Transaction;
+
+    // receiver is the same as the sender
+    create_transaction_for_setting_new_uris({
+        sender: IAddress;
+        tokenIdentifier: string;
+        tokenNonce: int;
+        uris: List[string]
+    }): Transaction;
+
+    // receiver is the same as the sender
+    create_transaction_for_modifying_creator({
+        sender: IAddress;
+        tokenIdentifier: string;
+        tokenNonce: int;
+    }): Transaction;
+
+    // receiver is the same as the sender
+    create_transaction_for_updating_metadata({
+        sender: IAddress;
+        tokenIdentifier: string;
+        tokenName?: string;
+        royalties?: int;
+        hash?: string;
+        attributes?: bytes;
+        uris?: List[string];
```

Comment:
As above, prefix with `new_` to ensure there is no ambiguity? E.g. so that it's obvious that the token identifier isn't changed, but the rest of the fields are.

Reply:
done